tensorflow_model_optimization/g3doc/guide/clustering/clustering_comprehensive_guide.ipynb

```diff
@@ -0,0 +1,536 @@
+{
```

Comment:
<p>1.<code>cluster_weights</code>  doesn't have a notion of a clustering schedule yet. This comment is not relevant yet.</p><p> </p><p>2. "To find the optimal number of clusters per layer, empirical methods can be used and generally the first few layers are more sensitive to clustering than the later ones; the layers with more parameters are more forgiving to aggressive clustering": this is already stated above, so I had something like "For general ideas to improve model accuracy, look for tips for your use case(s) under "Define model". There are other tips also besides clustering later ones being better that they should look at, if they miss it. You should also have this tip to cluster layers with more parameters as opposed to fewer parameters up there (it's relevant for deciding which layer to cluster in "Cluster some layers").</p><p> </p><p>3. <code>tf.keras.layers.DepthwiseConv2D</code>&nbsp;layers are not suited for clustering. Same as 2. I'd put this as a tip under "Cluster all layers" and point to "Cluster some layers" as a means of skipping DepthwiseConv2D.</p><p>Alternatively, it this is really the case, I'd think it'd be a better default in the clustering_registry to just not cluster DepthwiseConv2D. They don't have to consider this tip then. </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/model-optimization/pull/330/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
<blockquote>Alternatively, it this is really the case, I'd think it'd be a better default in the clustering_registry to just not cluster DepthwiseConv2D. They don't have to consider this tip then.</blockquote><p>No idea whether there will be users only want the best compression rate and ignore accuracy? Perhaps we can leave a warning in the API for clustering DepthwiseConv2D?</p>

Reply:
<p>DepthwiseConv2D was resolved in one of the PRs.</p>